pkg/standalone-utils/contracts/WeightedLPOracle.sol

```diff
@@ -0,0 +1,280 @@
+     */
+    function description() external view returns (string memory) {
+        return _description;
+    }
+
+    /**
+     * @notice Get data about a specific round, using the roundId.
+     * @dev This is unused, and always returns all zeros.
+     * @return roundId The round ID
+     * @return answer The answer for this round
+     * @return startedAt Timestamp when the round started
+     * @return updatedAt Timestamp when the round was updated
+     * @return answeredInRound [Deprecated] - Previously used when answers could take multiple rounds to be computed
+     */
+    function getRoundData(
+        uint80 /* _roundId */
+    )
+        external
+        pure
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        return (0, 0, 0, 0, 0);
+    }
+
+    /**
+     * @notice Get the data from the latest round.
+     * @return roundId The round ID
+     * @return answer The answer for this round
+     * @return startedAt Timestamp when the round started
+     * @return updatedAt Timestamp when the round was updated
+     * @return answeredInRound [Deprecated] - Previously used when answers could take multiple rounds to be computed
+     */
+    function latestRoundData()
+        external
+        view
+        returns (uint80 roundId, int256 answer, uint256 startedAt, uint256 updatedAt, uint80 answeredInRound)
+    {
+        (int256[] memory prices, uint256 _updatedAt) = _getFeedData();
+
+        uint256 tvl = _calculateTVL(prices);
+        uint256 totalSupply = IERC20(address(pool)).totalSupply();
+
+        uint256 lpPrice = tvl.divDown(totalSupply);
+
+        return (0, lpPrice.toInt256(), 0, _updatedAt, 0);
+    }
+
+    function _calculateTVL(int256[] memory prices) internal view returns (uint256 tvl) {
+        uint256 totalTokens = _totalTokens;
+
+        uint256[] memory weights = _getWeights(totalTokens);
+        (, , , uint256[] memory lastBalancesLiveScaled18) = _vault.getPoolTokenInfo(address(pool));
+
+        /**********************************************************************************************
+        // invariant                   _____                                                         //
+        // wi = weight index i          | |           wi                                             //
+        // pi = price index i      k *  | |  (pi/wi) ^   = tvl                                       //
+        // k = invariant                                                                             //
+        **********************************************************************************************/
+
+        uint256 k = pool.computeInvariant(lastBalancesLiveScaled18, Rounding.ROUND_UP);
+
+        tvl = FixedPoint.ONE;
+        for (uint256 i = 0; i < totalTokens; i++) {
+            tvl = tvl.mulUp(prices[i].toUint256().divDown(weights[i]).powUp(weights[i]));
+        }
+
+        tvl = tvl.mulUp(k);
```

Comment:
Shouldn't we round down here? Why would we inflate the TVL calculation with the rounding?